libraries/botbuilder-adapter-twitter/src/twitter.ts

```
@@ -0,0 +1,195 @@
+        }
+
+        const activity: Partial<Activity> = {
+            id: message.id,
+            timestamp: new Date(),
+            channelId: this.channel,
+            conversation: {
+                id: message.id,
+                isGroup: false,
+                conversationType: null,
+                tenantId: null,
+                name: ''
+            },
+            from: {
+                id: message.user.id,
+                name: message.user.screen_name
+            },
+            recipient: {
+                id: message.in_reply_to_screen_name,
+                name: message.in_reply_to_screen_name
+            },
+            text: message.text,
+            channelData: message,
+            localTimezone: null,
+            callerId: null,
+            serviceUrl: null,
+            listenFor: null,
+            label: message.id,
+            valueType: null,
+            type: null
+        };
+
+        if (activity.type === ActivityTypes.Message) {
+
+            if(message.entities.media != null && message.entities.media.length > 0) {
+                const media = message.entities.media;
+                activity.attachments = [];
+                for (const medium of media) {
+                    let contentType: string;
+                    /*
+                     * Likely need to do a little more parsing of the content/mime type.
+                     */
+                    switch(medium.type) {
+                        case 'photo':
+                            contentType = 'image/png';
+                            break;
+                        case 'video':
+                            contentType = 'video/mpeg';
+                            break;
+                        case 'animated_gif':
+                            contentType = 'image/gif';
+                            break;
+                        default:
+                            throw new Error(`Medium doesn't have an associated content type.`);
```

Do we want to throw an error or keep `contentType` empty or as 'unknown' or something.

---

Unsure.

---

Removed throwing exception in default.